File: src/main/java/alfheim/common/item/equipment/bauble/ItemCreativeReachPendant.java
package alfheim.common.item.equipment.bauble;

import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.item.ItemStack;
import net.minecraft.util.StatCollector;
import net.minecraftforge.client.event.RenderPlayerEvent;
import vazkii.botania.client.core.helper.IconHelper;
import vazkii.botania.common.Botania;

public class ItemCreativeReachPendant extends ItemPendant {

	public ItemCreativeReachPendant() {
		super("CreativeReachPendant");
	}

	@Override
	public void onEquippedOrLoadedIntoWorld(ItemStack stack, EntityLivingBase player) {
		Botania.proxy.setExtraReach(player, 100F);
	}

	@Override
	public void onUnequipped(ItemStack stack, EntityLivingBase player) {
		Botania.proxy.setExtraReach(player, -100F);
	}

	@Override
	public String getUnlocalizedNameInefficiently(ItemStack stack) {
		String s = this.getUnlocalizedName(stack);
		return s == null ? "" : StatCollector.translateToLocal(s);
	}
	
	@Override
	public void onPlayerBaubleRender(ItemStack stack, RenderPlayerEvent event, RenderType type) {
		// NO-OP
	}
	
	@Override
	public void registerIcons(IIconRegister reg) {
		itemIcon = IconHelper.forItem(reg, this);
	}
}

File: src/main/java/alfheim/common/item/equipment/bauble/ItemCreativeReachPendant.kt
package alfheim.common.item.equipment.bauble

import net.minecraft.client.renderer.texture.IIconRegister
import net.minecraft.entity.EntityLivingBase
import net.minecraft.item.ItemStack
import net.minecraft.util.StatCollector
import net.minecraftforge.client.event.RenderPlayerEvent
import vazkii.botania.client.core.helper.IconHelper
import vazkii.botania.common.Botania

class ItemCreativeReachPendant: ItemPendant("CreativeReachPendant") {
	
	override fun onEquippedOrLoadedIntoWorld(stack: ItemStack?, player: EntityLivingBase?) {
		Botania.proxy.setExtraReach(player, 100f)
	}
	
	override fun onUnequipped(stack: ItemStack?, player: EntityLivingBase?) {
		Botania.proxy.setExtraReach(player, -100f)
	}
	
	override fun getUnlocalizedNameInefficiently(stack: ItemStack): String {
		val s = this.getUnlocalizedName(stack)
		return if (s == null) "" else StatCollector.translateToLocal(s)
	}
	
	override fun onPlayerBaubleRender(stack: ItemStack, event: RenderPlayerEvent, type: IBaubleRender.RenderType) {
		// NO-OP
	}
	
	override fun registerIcons(reg: IIconRegister) {
		itemIcon = IconHelper.forItem(reg, this)
	}
}
